### PR TITLE
chore: Update README.md and make-release.sh

### PR DESCRIPTION
**Pull request by mkuznyetsov**
Signed-off-by: Mykhailo Kuznietsov <mkuznets@redhat.com>

### What does this PR do?
Populate the README.md with the basic information about the project, with brief overview of the structure, and CI. Also fix outdated references to CI scripts

### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/20492

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [x] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,34 @@
...
-### Che server
\ No newline at end of file
+[![Next build Status](https://github.com/eclipse-che/che-theia/actions/workflows/next-build.yml/badge.svg)]
+[![Release build Status](https://github.com/eclipse-che/che-theia/actions/workflows/release.yml/badge.svg)]
+
+# What is Che server
+Che Server is a core component of the Eclipse Che. The component is responsibe for creation and managing of Che workspaces.
+
+# Project structure
+Che Server is mostly a Java web application deployed on a Apache Tomcat server in a container. The primary root module ['pom.xml'](https://github.com/eclipse-che/che-server/tree/main/pom.xml) lists all the java submodules. In the ['dockerfiles'](https://github.com/eclipse-che/che-server/tree/main/dockerfiles) directory contains container image for Che Server, as well as additional images, such as container for PostgreSQL DB. Additionally, there is a separate ['typescript-dto'](https://github.com/eclipse-che/che-server/tree/main/typescript-dto) module, that provides DTO objects for typescript projects that may depend on Che Server, such as Che Theia.
```
Comment (sparkoo) [inline README.md:8]: Instead of one long paragraph, I would maybe write it as bullet list like CI part

File: README.md [hunk 2]
```diff
@@ -1,2 +1,34 @@
-TODO
-### Che server
\ No newline at end of file
+[![Next build Status](https://github.com/eclipse-che/che-theia/actions/workflows/next-build.yml/badge.svg)]
+[![Release build Status](https://github.com/eclipse-che/che-theia/actions/workflows/release.yml/badge.svg)]
+
+# What is Che server
+Che Server is a core component of the Eclipse Che. The component is responsibe for creation and managing of Che workspaces.
```
Comment (sparkoo) [inline README.md:5]: maybe link "Eclipse Che" to main Che repo or Eclipse Che webpage?

File: README.md (hunk 1 above)
Comment (sparkoo) [reply README.md:8]: especially sentence about dockerfiles feels clumsy. the directory definitely does not containe conainer images. It contains only dockerfiles to build these images.

File: README.md [hunk 3]
```diff
@@ -1,2 +1,34 @@
...
+
+# Project structure
+Che Server is mostly a Java web application deployed on a Apache Tomcat server in a container. The primary root module ['pom.xml'](https://github.com/eclipse-che/che-server/tree/main/pom.xml) lists all the java submodules. In the ['dockerfiles'](https://github.com/eclipse-che/che-server/tree/main/dockerfiles) directory contains container image for Che Server, as well as additional images, such as container for PostgreSQL DB. Additionally, there is a separate ['typescript-dto'](https://github.com/eclipse-che/che-server/tree/main/typescript-dto) module, that provides DTO objects for typescript projects that may depend on Che Server, such as Che Theia.
+
+# Build requirements
+- Apache Maven 3.6.3 or Higher
+- JDK Version 11
+- Docker
```
Comment (sparkoo) [inline README.md:13]: Or Podman, although you can't run integration tests with podman.

**Review by sparkoo — COMMENTED** [review]
thank you for this! just few suggestions. Also how to build and debug sections would be nice.

**Comment by nickboldt** [comment]
> thank you for this! just few suggestions. Also how to build and debug sections would be nice.

Could you add those sections to the doc and push a commit into this PR?

File: README.md [hunk 4]
```diff
@@ -1,2 +1,40 @@
...
+
+# What is Che server
+Che Server is a core component of the [Eclipse Che](https://github.com/eclipse/che/). This component is responsibe for creation and managing of Che workspaces, but will some day be replaced by the [Dev Workspace Operator](https://github.com/devfile/devworkspace-operator). 
+
+# Project structure
+Che Server is primarily a Java web application deployed on a Apache Tomcat server in a container. 
+
+The primary root module ['pom.xml'](https://github.com/eclipse-che/che-server/tree/main/pom.xml) lists all the java submodules. 
+
+The ['dockerfiles'](https://github.com/eclipse-che/che-server/tree/main/dockerfiles) directory contains container image for Che Server, as well as additional images, such as container for PostgreSQL DB.
```
Comment (sparkoo) [inline README.md:12]: This description is weird, there are no container images, there are only dockerfiles to build the images.

**Comment by sparkoo** [comment]
> > thank you for this! just few suggestions. Also how to build and debug sections would be nice.
> 
> were you thinking someting more complicated than `mvn clean install` or `mvn clean install -X` ?
> 
> Could you add those sections to the doc and push a commit into this PR?

I don't need anything more complicated. Feel free to merge without it, I don't have cycles for it atm.

File: README.md (hunk 4 above)
Comment (mkuznyetsov) [reply README.md]: will fix

File: README.md (hunk 1 above)
Comment (mkuznyetsov) [reply README.md]: reworked

**Pull request merged**